### PR TITLE
Add examples + documentation for std::path

### PR DESCRIPTION
**Pull request by steveklabnik**

### CODE REVIEW

File: src/libstd/path.rs [hunk 1]
```diff
@@ -1207,12 +1238,28 @@ impl Path {
     /// Directly wrap a string slice as a `Path` slice.
     ///
     /// This is a cost-free conversion.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::path::Path;
+    ///
+    /// Path::new("foo.txt");
```
Comment (FlaPer87) [inline src/libstd/path.rs]: nit: In other places you bind this to a variable. I guess we should do it here as well, for consistency with other examples.

**Comment by FlaPer87** [comment]
tiny nit, r=me if you either think it's not worth fixing it or you fix it. :cake:

File: src/libstd/path.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libstd/path.rs]: I did it for those because those return a `Result<T, E>`, which is `must_use`.

Comment (FlaPer87) [inline src/libstd/path.rs]: I knew there was a reason! Thanks!

**Pull request merged**